core/runtime/src/main/java/io/quarkus/runtime/configuration/Services.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.runtime.configuration;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class Services {
+    private static final Map<String, Service> services = new ConcurrentHashMap<>();
+
+    private Services() {
+        throw new UnsupportedOperationException();
+    }
+
+    public static Optional<Service> get(String name) {
+        return Optional.ofNullable(services.get(name));
+    }
+
+    public static void register(final String service, final String protocol, final String host, final int port) {
+        services.put(service, new Service(protocol, host, port));
+    }
+
+    public static class Service {
+        private final String protocol;
+        private final String host;
+        private final int port;
+
+        public Service(final String protocol, final String host, final int port) {
```

Comment:
As I already said, this is very close to Stork, and I think, it can be confusing. 

Stork also has services, with a very similar API (with `secure` as additional field indicating if TLS needs to be used or not).